x/gov/simulation/operations.go

```diff
@@ -0,0 +1,282 @@
+		}
+
+		proposalID = uint64(math.Max(float64(proposalID)-1, 0))
+
+		// 2) Schedule operations for votes
+		// 2.1) first pick a number of people to vote.
+		curNumVotesState = numVotesTransitionMatrix.NextState(r, curNumVotesState)
+		numVotes := int(math.Ceil(float64(len(accs)) * statePercentageArray[curNumVotesState]))
+
+		// 2.2) select who votes and when
+		whoVotes := r.Perm(len(accs))
+
+		// didntVote := whoVotes[numVotes:]
+		whoVotes = whoVotes[:numVotes]
+		votingPeriod := k.GetVotingParams(ctx).VotingPeriod
+
+		fops := make([]simulation.FutureOperation, numVotes+1)
+		for i := 0; i < numVotes; i++ {
+			whenVote := ctx.BlockHeader().Time.Add(time.Duration(r.Int63n(int64(votingPeriod.Seconds()))) * time.Second)
+			fops[i] = simulation.FutureOperation{
+				BlockTime: whenVote,
+				Op:        operationSimulateMsgVote(ak, k, accs[whoVotes[i]], int64(proposalID)),
+			}
+		}
+
+		// 3) Make an operation to ensure slashes were done correctly. (Really should be a future invariant)
+		// TODO: Find a way to check if a validator was slashed other than just checking their balance a block
+		// before and after.
+
+		return opMsg, fops, nil
+	}
+}
+
+// SimulateTextProposalContent returns random text proposal content.
+func SimulateTextProposalContent(r *rand.Rand, _ sdk.Context, _ []simulation.Account) types.Content {
+	return types.NewTextProposal(
+		simulation.RandStringOfLength(r, 140),
+		simulation.RandStringOfLength(r, 5000),
+	)
+}
+
+// SimulateMsgDeposit generates a MsgDeposit with random values.
+func SimulateMsgDeposit(ak types.AccountKeeper, k keeper.Keeper) simulation.Operation {
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account) (
+		opMsg simulation.OperationMsg, fOps []simulation.FutureOperation, err error) {
+
+		acc := simulation.RandomAcc(r, accs)
+		proposalID, ok := randomProposalID(r, k, ctx)
+		if !ok {
+			return simulation.NoOpMsg(types.ModuleName), nil, nil
+		}
+
+		deposit, err := randomDeposit(r, ctx, ak, k, acc.Address)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
+
+		msg := types.NewMsgDeposit(acc.Address, proposalID, deposit)
+
+		fromAcc := ak.GetAccount(ctx, acc.Address)
+		fees, err := helpers.RandomFees(r, ctx, fromAcc, deposit)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
+
+		tx := helpers.GenTx(
+			[]sdk.Msg{msg},
+			fees,
+			[]uint64{fromAcc.GetAccountNumber()},
+			[]uint64{fromAcc.GetSequence()},
+			[]crypto.PrivKey{acc.PrivKey}...,
+		)
+
+		res := app.Deliver(tx)
+		if !res.IsOK() {
+			return simulation.NoOpMsg(types.ModuleName), nil, errors.New(res.Log)
+		}
+
+		return simulation.NewOperationMsg(msg, true, ""), nil, nil
+	}
+}
+
+// SimulateMsgVote generates a MsgVote with random values.
+func SimulateMsgVote(ak types.AccountKeeper, k keeper.Keeper) simulation.Operation {
+	return operationSimulateMsgVote(ak, k, simulation.Account{}, -1)
```

Comment:
where if the function used and I noticed it no longer uses a default proposal ID of `0` but generates a random one now which makes sense, but maybe this should be noted in the comments?